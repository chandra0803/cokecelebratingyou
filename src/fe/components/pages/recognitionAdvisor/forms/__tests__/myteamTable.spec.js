/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import MyteamTable from '../myteamTable.js';
import renderer from 'react-test-renderer';

window.recognitionAdvisor = {
  raUrl: 'ra/reminders.action',
  raTilePageDisplay: 'yes',
  raDetailPageDisplay: 'no',
  raEndModelPageDisplay: 'no',
  content: []
};

const defaultProps = {
    recognitionAdvisor: {
      raUrl: 'ra/reminders.action',
      raTilePageDisplay: 'yes',
      raDetailPageDisplay: 'no',
      raEndModelPageDisplay: 'no',
      content: {
        'recognition.content.model.ra_days': 'days',
        'recognition.content.model.info.ra_day': 'day',
        'recognition.content.model.info.ra_today': 'Today',
        'recognition.content.model.info.ra_pending': 'pending',
        'recognition.content.model.info.ra_pending_message': 'pending message',
        'recognition.content.model.info.ra_status': 'status',
        'recognition.content.model.info.ra_point': 'point',
        'recognition.content.model.info.ra_points': 'points'
      }
    }
};

it( 'renders the dom like the snapshot when the props have default values.', () => {
    const content = {
      content: {
        'recognition.content.model.ra_days': 'days',
        'recognition.content.model.info.ra_day': 'day',
        'recognition.content.model.info.ra_today': 'Today',
        'recognition.content.model.info.ra_pending': 'pending',
        'recognition.content.model.info.ra_pending_message': 'pending message',
        'recognition.content.model.info.ra_status': 'status',
        'recognition.content.model.info.ra_point': 'point',
        'recognition.content.model.info.ra_points': 'points'
      }
    };
    const props = {
        ...defaultProps,
        ...content,
        recognitionAdvisorFunc: jest.fn(),
        raEligibleProgramsFunc: jest.fn()
    };
    const component = renderer.create(
        <div>
            <MyteamTable { ...props } />
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
