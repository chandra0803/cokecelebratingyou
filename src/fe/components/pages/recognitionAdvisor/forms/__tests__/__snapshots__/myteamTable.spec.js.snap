// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div
    className="myteamTable"
  >
    <table
      className="table table-striped"
    >
      <thead>
        <tr>
          <th
            className="active"
            onClick={[Function]}
          >
            <a
              data-sort="is_new_hire"
              data-sortBy="desc"
              href="#"
              onClick={[Function]}
            >
              status
               
            </a>
             
            <i
              className="icon-arrow-1-down"
            />
          </th>
          <th
            className="null"
            onClick={[Function]}
          >
            <a
              data-sort="first_name"
              data-sortBy="asc"
              href="#"
              onClick={[Function]}
            />
             
            <i
              className="icon-arrow-1-up"
            />
          </th>
          <th
            className="null"
            onClick={[Function]}
          >
            <a
              data-sort="by_me_date_sent"
              data-sortBy="asc"
              href="#"
              onClick={[Function]}
            />
             
            <i
              className="icon-arrow-1-up"
            />
          </th>
          <th
            className="null"
            onClick={[Function]}
          >
            <a
              data-sort="by_oth_date_sent"
              data-sortBy="asc"
              href="#"
              onClick={[Function]}
            />
             
            <i
              className="icon-arrow-1-up"
            />
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
